src/background/stores/card-selection-store.ts

```diff
@@ -24,17 +24,72 @@ export class CardSelectionStore extends BaseStoreImpl<CardSelectionStoreData> {
 
     public getDefaultState(): CardSelectionStoreData {
         const defaultValue: CardSelectionStoreData = {
-            rules: null,
+            rules: {},
         };
 
         return defaultValue;
     }
 
-    public toggleRuleExpandCollapse(payload: RuleExpandCollapsePayload): void {}
+    private deselectAllCardsInRule = (rule: RuleExpandCollapseData): void => {
+        if (!rule) {
+            return;
+        }
 
-    public toggleCardSelection(payload: CardSelectionPayload): void {}
+        for (const resultInstanceUid of Object.keys(rule.cards)) {
+            rule.cards[resultInstanceUid] = false;
+        }
+    };
 
-    public CollapseAllRules(): void {}
+    public toggleRuleExpandCollapse = (payload: RuleExpandCollapsePayload): void => {
+        if (!payload || !payload.ruleId || !this.state.rules || !this.state.rules[payload.ruleId]) {
```

Comment:
The condition for "!this.state.rules" will always be true as the default is not null.

(Not sure if you're aware but an empty object will evaluate to true.)

Reply:
Having this check allowed me to find the problem where the functions needed to be declared as arrow functions. So I feel somewhat validated for adding it. Thoughts?

Reply:
We usually do not put stuff like this, that validates whether something is an arrow function or otherwise, in our code, so I'd err on the side on not including it but up to you.

Might actually make sense to discuss briefly in an engineering discussion if you feel strongly otherwise.

Reply:
Made the change.